### ISSUE TITLE
Cannot open remote Jupyter notebooks with Markdown

### ISSUE THREAD

**Issue by DoWalther** [microsoft/vscode#126929]
## Environment data

-   VS Code version: 1.57.0 (Universal)
-   Jupyter Extension version (available under the Extensions sidebar): 2021.6.999
-   Python Extension version (available under the Extensions sidebar): 2021.6.0-rc
-   OS (Mac) and version: Darwin x64 20.3.0 - 20.5.0 (tested)
-   Python and/or Anaconda version: conda 4.10.0
-   Type of virtual environment used (virtualenv): Kubernetes Pod
-   Jupyter server running: Remote 

## Expected behaviour
Jupyter Notebooks should open normal and mardown cells should be shown.

## Actual behaviour
Creating new Notebooks works fine, when writing Markdown cells and execute them, the markdown cell disappears. When closing and reopening the Notebook, it loads infinte without any error message.

## Steps to reproduce:
1. Create a new notebook and write a markdown cell
2. Execute the markdown cell --> Markdown disappears
3. Save the notebook and reopen it.

![github_issue](https://user-images.githubusercontent.com/7305011/122539915-695e7780-d028-11eb-85ad-ea54cf28c760.gif)

This issue is resolved when disabeling the new extensible markdown renderer in the user settings.

**Comment by rchiodo** [comment]
From your output it looks like this is a markdown issue.

**Comment by roblourens** [comment]
Can you check for errors in the devtools console? (Help > Toggle Developer Tools)

**Comment by rchiodo** [comment]
@roblourens I believe that's in the collapsed section above.

```
notebookWebviewPreloads.js:3 Uncaught (in promise) TypeError: Failed to execute 'fetch' on 'Window': Failed to parse URL 
from https://vscode-remote%2Bk8s-container%2Bcontext%3Ddwalther%2Bpodname%3Ddwa-
conda%2Bnamespace%3Dresearch%2Bname%3Ddwalther-tf%2Bimage%3Dfoo-bar.de%253a30500%252fconda-
kite%253a1.01.vscode-resource.vscode-webview.net/home/dwalther/.vscode-server
/bin/507ce72a4466fbb27b715c3722558bb15afa9f48/extensions/markdown-language-features/notebook-out/index.js
  at s (notebookWebviewPreloads.js:3)
  at c (notebookWebviewPreloads.js:3)
  at _load (notebookWebviewPreloads.js:3)
  at _.load (notebookWebviewPreloads.js:3)
  at notebookWebviewPreloads.js:3
  at Array.map (<anonymous>)
  at Object.renderMarkdown (notebookWebviewPreloads.js:3)
  at J (notebookWebviewPreloads.js:3)
  at U (notebookWebviewPreloads.js:3)
  at notebookWebviewPreloads.js:3
```

**Comment by roblourens** [comment]
@chrmarti is this your remote authority?
 
It seems like vscode is more permissive in what characters are allowed in the authority than the actual runtime. Seems that an escaped percent sign is not valid in an authority? I thought it would be allowed if percent-encoded

![image](https://user-images.githubusercontent.com/323878/123017083-bd4ac280-d380-11eb-8e17-39abe27b308d.png)

Compare with other symbols

![image](https://user-images.githubusercontent.com/323878/123017322-36e2b080-d381-11eb-9727-9aa481192a92.png)

**Comment by chrmarti** [comment]
Yes, that's a remote URI for attaching to a Kubernetes pod. Are the percent characters added on your side @roblourens? (They look like separators, but could be part of the container image name.)

**Comment by roblourens** [comment]
No, the full remote authority is apparently this `"vscode-remote+k8s-container+context=dwalther+podname=dwa-conda+namespace=research+name=dwalther-tf+image=foo-bar.de%3a30500%2fconda-kite%3a1"` is the `+context=` added by your extension? I guess this is also a double-escaping thing, a percent-encoded string is percent-encoded a second time when it's put into the authority

**Comment by chrmarti** [comment]
I'm encoding each value with `encodeURIComponent()` before passing the authority to the 'vscode.newWindow' command as `remoteAuthority`. @aeschli Should I leave the encoding to the command?

Yes, context is added by Remote-Containers.

**Comment by aeschli** [comment]
We use the remoteAuthority as authority when constructing the URI. Encoding happens there.

**Comment by daniel-scheliga** [comment]
I guess this problem (https://github.com/microsoft/vscode/issues/126845) may be related. 
Getting similar error in the Developer Output when I try to render plots with pyplot.
```
notebookWebviewPreloads.js:3 TypeError: Failed to execute 'fetch' on 'Window': Failed to parse URL from https://vscode-remote%2Bk8s-container%2Bcontext%3Ddscheliga-lp%2Bpodname%3Ddsc-conda-lp%2Bnamespace%3DLP%2Bname%3Ddscheliga-tf%2Bimage%3Dfoo-bar.de%253a30500%252fconda-kite%253a1.01.vscode-resource.vscode-webview.net/home/dscheliga/.vscode-server/extensions/ms-toolsai.jupyter-2021.6.999662501/out/client_renderer/renderers.js
    at s (notebookWebviewPreloads.js:3)
    at c (notebookWebviewPreloads.js:3)
    at _load (notebookWebviewPreloads.js:3)
    at _.load (notebookWebviewPreloads.js:3)
    at Object.load (notebookWebviewPreloads.js:3)
    at notebookWebviewPreloads.js:3
    at q.outputs.set.queue (notebookWebviewPreloads.js:3)
    at new Promise (<anonymous>)
    at Object.enqueue (notebookWebviewPreloads.js:3)
    at notebookWebviewPreloads.js:3
```
The notebooks itself is still loading, but the plots are not rendered (instead the error is displayed)
